uvicorn/config.py

```diff
@@ -119,43 +125,43 @@ class Config:
     def __init__(
         self,
         app,
```

Comment:
`app` will need an annotation.

I agree that `ASGIApplication`, from `uvicorn._types`, would be suitable here. As mentioned in the related PR encode/uvicorn#990, it's a bit challenging, because `app` is sometimes also a string. Annotating as `Union[ASGIApplication, str]` should address these use cases. If not, relaxing the annotation with `Union[Callable, str]` could also work.

`app: Union[ASGIApplication, str]`